android/lib/src/main/java/com/swmansion/gesturehandler/PanGestureHandler.kt

```diff
@@ -244,14 +244,17 @@ class PanGestureHandler(context: Context?) : GestureHandler<PanGestureHandler>()
       if (shouldFail()) {
         fail()
       } else if (shouldActivate()) {
-        // reset starting point
-        startX = lastX
-        startY = lastY
         activate()
       }
     }
   }
 
+  override fun beforeActivation() {
```

Comment:
this seems like something we should be doing in "reset" logic, no? Why has this been moved here?

Reply:
When the pan activates it has to "reset" its starting position so the translation is calculated relative to the activation point and not the point where the finger was first placed on the screen. Before it was called just before the `activate` in `onHandle` method.